### ISSUE TITLE
Build warnings

### ISSUE THREAD

**Issue by seahawk1986** [j1rie/vdr-plugin-statusleds#1]
The plugin gives me various build warnings including writing out of bounds:
````
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:547:9: warning: ignoring return value of 'int system(const char*)' declared with attribute 'warn_unused_result' [-Wunused-result]
  547 |   system(Cmd);
      |   ~~~~~~^~~~~
statusleds.c: In member function 'virtual void cRecordingPresignal::Action()':
statusleds.c:583:20: warning: ignoring return value of 'ssize_t write(int, const void*, size_t)' declared with attribute 'warn_unused_result' [-Wunused-result]
  583 |               write(iConsole, "\007", 1);
      |               ~~~~~^~~~~~~~~~~~~~~~~~~~~
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:541:35: warning: '%s' directive writing up to 255 bytes into a region of size 85 [-Wformat-overflow=]
  541 |   sprintf(Cmd, "blink %s -r %d -m %s", Leds[iLed],
      |                                   ^~
  542 |                                 forceOff ? 0 : iRecordings, sBlinkdHost);
      |                                                             ~~~~~~~~~~~
In file included from /usr/include/stdio.h:980,
                 from /usr/include/vdr/i18n.h:13,
                 from statusleds.c:8:
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:541:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output 16 or more bytes (assuming 271) into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:545:19: warning: ' &' directive writing 2 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  545 |   sprintf(Cmd, "%s &", Cmd);
      |                   ^~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:545:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 3 and 102 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
statusleds.c:544:21: warning: ' -t ' directive writing 4 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  544 |     sprintf(Cmd, "%s -t %d", Cmd, iBlinkdPort);
      |                     ^~~~
statusleds.c:544:18: note: directive argument in the range [1, 2147483647]
  544 |     sprintf(Cmd, "%s -t %d", Cmd, iBlinkdPort);
      |                  ^~~~~~~~~~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:544:12:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 6 and 114 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
statusleds.c:545:19: warning: ' &' directive writing 2 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  545 |   sprintf(Cmd, "%s &", Cmd);
      |                   ^~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:545:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 3 and 102 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
statusleds.c:541:35: warning: '%s' directive writing up to 255 bytes into a region of size 85 [-Wformat-overflow=]
  541 |   sprintf(Cmd, "blink %s -r %d -m %s", Leds[iLed],
      |                                   ^~
  542 |                                 forceOff ? 0 : iRecordings, sBlinkdHost);
      |                                                             ~~~~~~~~~~~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:541:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output 16 or more bytes (assuming 271) into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:545:19: warning: ' &' directive writing 2 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  545 |   sprintf(Cmd, "%s &", Cmd);
      |                   ^~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:545:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 3 and 102 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
statusleds.c:544:21: warning: ' -t ' directive writing 4 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  544 |     sprintf(Cmd, "%s -t %d", Cmd, iBlinkdPort);
      |                     ^~~~
statusleds.c:544:18: note: directive argument in the range [1, 2147483647]
  544 |     sprintf(Cmd, "%s -t %d", Cmd, iBlinkdPort);
      |                  ^~~~~~~~~~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:544:12:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 6 and 114 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
statusleds.c:545:19: warning: ' &' directive writing 2 bytes into a region of size between 1 and 100 [-Wformat-overflow=]
  545 |   sprintf(Cmd, "%s &", Cmd);
      |                   ^~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'void Blinkd(bool)' at statusleds.c:545:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 3 and 102 bytes into a destination of size 100
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/bits/stdio2.h:30:34: warning: '__builtin___sprintf_chk' argument 5 overlaps destination object 'Cmd' [-Wrestrict]
statusleds.c: In function 'void Blinkd(bool)':
statusleds.c:539:8: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  539 |   char Cmd[100];
      |        ^~~
````

**Comment by j1rie** [comment]
What do I have to do, in order to see those warnings?
When I build in my environment, I don't see them.

I don't use blinkd, and I even don't know, if it still exists.
Didn't find it neither on google nor on github nor on rpmsearch.
Maybe I should remove the relating code?

**Comment by seahawk1986** [comment]
I get those warnings when building the packages on Launchpad from Ubuntu 20.04 to Ubuntu 24.04, the compiler call for the file is `g++ -g -O3 -Wall -Werror=overloaded-virtual -Wno-parentheses -fPIC -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -c -DPLUGIN_NAME_I18N='"statusleds"' -D_GNU_SOURCE  -o statusleds.o statusleds.c`

- https://launchpadlibrarian.net/749038531/buildlog_ubuntu-focal-amd64.vdr-plugin-statusleds_0.4+git20240912-12-a2cc1b3-0yavdr0~focal_BUILDING.txt.gz
- https://launchpadlibrarian.net/749038152/buildlog_ubuntu-jammy-amd64.vdr-plugin-statusleds_0.4+git20240912-12-a2cc1b3-0yavdr0~jammy_BUILDING.txt.gz
- https://launchpadlibrarian.net/749039437/buildlog_ubuntu-noble-arm64.vdr-plugin-statusleds_0.4+git20240912-12-a2cc1b3-0yavdr0~noble_BUILDING.txt.gz (the amd64 build will follow, today several builds failed without a clear reason on LaunchPad)

blinkd seems to be quite old - in the old debian package for the statusleds plugin it's only referenced as a suggested dependency: https://salsa.debian.org/vdr-team/vdr-plugin-statusleds/-/blob/master/debian/control?ref_type=heads#L14 - this was the project site (unfortunately no source code): https://www.nongnu.org/blinkd/ and the package has been removed from the debian repositories around the release of Ubuntu 9.10 - in https://www.vdr-portal.de/forum/index.php?thread/124008-howto-blinkd-und-statusleds-ersatz/ someone posted a python replacement script.

**Comment by j1rie** [comment]
I guess my compiler does not show those warnings, because it's too old (opensuse leap 15.6 has gcc 7).

I've removed blinkd and fixed a few things in version 0.5.
